src/server.py

```diff
@@ -78,6 +81,37 @@ def get_date_range(dates):
                 'endDate': max(dates).strftime('%Y-%m-%d %H:%M')}
 
 
+def valve(head_rev, pushlog_id):
+    """Determine which kind of job should been returned"""
+    low_job_return = True
+    COUNTER.increase_the_counter()
+    request_list = []
+    try:
+        request_list = session.query(TaskRequests.head_rev, TaskRequests.pushlog_id,
+                                     TaskRequests.low_job_return).all()
+    except Exception:
+        session.rollback()
```

Comment:
I still don't understand what rollback() is for, is this the database?  if we have an exception will the data we are working with be incorrect?  My recommendation of decreasing the counter was that if we have an exception we should decrease the counter- I think our failure plan is to return ALL jobs